pageserver/src/pgdatadir_mapping.rs

```diff
@@ -424,6 +424,12 @@ impl Timeline {
 
         let mut found_smaller = false;
         let mut found_larger = false;
+
+        // This fallpoint is used to check whether error propagation is handled correctly for cancelled request.
+        fail::fail_point!("timeline-request-cancelled", |_| {
+            Err(PageReconstructError::Cancelled)
+        });
+
```

Comment:
I think we could move this to the top of the fn body..?

IMHO the comment could be removed, there is no other reason for such failpoint to exist.